backend-api/src/functions/asyncActiveSession/asyncActiveSessionHandler.ts

```diff
@@ -26,63 +26,61 @@ export async function lambdaHandlerConstructor(
     event.headers["Authorization"] ?? event.headers["authorization"],
   );
   if (authorizationHeaderResult.isError) {
-    logger.log("AUTHENTICATION_HEADER_INVALID", {
+    logger.log("INVALID_AUTHENTICATION_HEADER", {
       errorMessage: authorizationHeaderResult.value.errorMessage,
     });
     return unauthorizedResponse;
   }
-
   const serviceTokenJwe = authorizationHeaderResult.value;
 
   const decryptResult = await dependencies
     .jweDecrypter(config.ENCRYPTION_KEY_ARN)
     .decrypt(serviceTokenJwe);
-
   if (decryptResult.isError) {
-    logger.log("JWE_DECRYPTION_ERROR", {
+    if (decryptResult.value.errorCategory === "CLIENT_ERROR") {
+      logger.log("JWE_DECRYPTION_ERROR", {
+        errorMessage: decryptResult.value.errorMessage,
+      });
+      return badRequestResponse("Failed to decrypt service token");
+    }
+    logger.log("INTERNAL_SERVER_ERROR", {
       errorMessage: decryptResult.value.errorMessage,
     });
-    return badRequestResponse("Failed decrypting service token JWE");
+    return serverErrorResponse;
   }
-
   const serviceTokenJwt = decryptResult.value;
 
-  const tokenServiceDependencies = dependencies.tokenServiceDependencies;
-  const tokenService = dependencies.tokenService(tokenServiceDependencies);
-  const getSubFromTokenResult = await tokenService.getSubFromToken(
-    config.STS_JWKS_ENDPOINT,
+  const tokenService = dependencies.tokenService(config.STS_JWKS_ENDPOINT);
+  const validateServiceTokenResult = await tokenService.validateServiceToken(
+    serviceTokenJwt,
     {
-      maxAttempts: 3,
-      delayInMillis: 100,
+      aud: config.AUDIENCE,
+      iss: config.STS_BASE_URL,
+      scope: "idCheck.activeSession.read",
     },
-    serviceTokenJwt,
   );
-
-  if (getSubFromTokenResult.isError) {
-    if (getSubFromTokenResult.value.errorCategory === "CLIENT_ERROR") {
-      logger.log("FAILED_TO_GET_SUB_FROM_SERVICE_TOKEN", {
-        errorMessage: getSubFromTokenResult.value.errorMessage,
+  if (validateServiceTokenResult.isError) {
+    if (validateServiceTokenResult.value.errorCategory === "CLIENT_ERROR") {
+      logger.log("SERVICE_TOKEN_VALIDATION_ERROR", {
+        errorMessage: validateServiceTokenResult.value.errorMessage,
       });
-      return badRequestResponse(getSubFromTokenResult.value.errorMessage);
+      return badRequestResponse("Failed to validate service token");
```

Comment:
I've decided that if the token claims are invalid or the signature is invalid, I will return the same response. The actual problem is logged on lines 64-65. If this isn't appropriate, please do let me know/let's have a chat.